### PR TITLE
Spelling

### PR DESCRIPTION
**Pull request by jsoref**
This PR corrects misspellings identified by the [check-spelling action](https://github.com/marketplace/actions/check-spelling).

The misspellings have been reported at https://github.com/jsoref/podman/commit/04624679014630f9d7d486be6262faac1fcacc39#commitcomment-45284288

The action reports that the changes in this PR would make it happy: https://github.com/jsoref/podman/commit/83328c1aa180d27ccc45ae030679eca30274f212

Note: this PR does not include the action. If you're interested in running a spell check on every PR and push, that can be offered separately.

---
I understand that changes.txt is generated from commit messages, but commit messages have typos, which make them fairly misleading. It's trivial for me to drop changes for this file if requested (or any other file(s)).

### CODE REVIEW

File: commands-demo.md
Comment (jsoref) [inline commands-demo.md:51]: This is a bugfix. The current url is 404.

File: docs/source/markdown/podman-network-reload.1.md
Comment (jsoref) [inline docs/source/markdown/podman-network-reload.1.md:12]: Both spellings are present, it's really confusing.

File: docs/source/markdown/podman-run.1.md
Comment (jsoref) [inline docs/source/markdown/podman-run.1.md:949]: I could be totally wrong about this. Maybe it should have been `are allowed`

File: libpod/kube.go
Comment (jsoref) [inline libpod/kube.go:406]: w/o this, tooling won't spot it...

File: pkg/rootless/rootless_linux.go
Comment (jsoref) [inline pkg/rootless/rootless_linux.go:260]: This is notable

File: test/e2e/images_test.go
Comment (jsoref) [inline test/e2e/images_test.go:159]: Probably a notable change

File: test/e2e/play_kube_test.go
Comment (jsoref) [inline test/e2e/play_kube_test.go:19]: Changes like this worry me.

Personally, I'd rather people use things like `_` to indicate intentionally wrong things.

File: test/system/065-cp.bats
Comment (jsoref) [inline test/system/065-cp.bats:502]: scary

File: test/system/070-build.bats
Comment (jsoref) [inline test/system/070-build.bats:348]: I understand this isn't a typical typo, but this repo doesn't have many items like this, so I'm treating it like one.

File: libpod/container_config.go
Comment (Luap99) [inline libpod/container_config.go:261]: @mheon PTAL

I feel like this might break aliases.

**Review by Luap99 — COMMENTED** [review]
Thanks for the PR @jsoref

`changelog.txt` should not be changed.

Also can you please squash your commits.

File: docs/source/markdown/podman-run.1.md
Comment (Luap99) [reply docs/source/markdown/podman-run.1.md:949]: yes it should be `are allowed`

File: libpod/container_config.go
Comment (jsoref) [reply libpod/container_config.go:261]: I'm going to wait to do anything until I see a response (the CI seems fairly expensive).

File: libpod/define/container_inspect.go
Comment (jsoref) [inline libpod/define/container_inspect.go:160]: @Luap99 Note that this is similar

File: libpod/container_config.go
Comment (mheon) [reply libpod/container_config.go:261]: Yeah, this is a breaking change to DB schema - need to leave as-is. Good catch @Luap99

File: libpod/define/container_inspect.go
Comment (mheon) [reply libpod/define/container_inspect.go:160]: This one is fine, though, `omitempty` is a language keyword, not directly part of the schema.

File: libpod/container_config.go
Comment (jsoref) [reply libpod/container_config.go:261]: Would you take a comment indicating it's a known botch? (I wouldn't include it in this PR, just asking for in general)

Comment (mheon) [reply libpod/container_config.go]: Sure, that's always welcome

Comment (jsoref) [reply libpod/container_config.go]: Any particular style to it?

**Comment by edsantiago** [comment]
Restarted one test failure, [int podman fedora32 root container](https://storage.googleapis.com/cirrus-ci-6707778565701632-fcae48/artifacts/containers/podman/5043977965535232/html/int-podman-fedora-32-root-container.log.html#t--podman-pod-rm-removes-a-pod-with-a-container--1), which failed with "error removing pod". That's not a known flake, but I also don't see any way this PR could've caused it.

File: contrib/cirrus/ext_svc_check.sh
Comment (edsantiago) [inline contrib/cirrus/ext_svc_check.sh:9]: The actual correction here should be `succeed`, see context in previous line

File: contrib/cirrus/logformatter
Comment (edsantiago) [inline contrib/cirrus/logformatter:443]: "envariable" has been standard UNIXspeak for decades. I've never seen nor heard of it with two 'v's; the usages I find on DDG all show it in mixed caps: 'envVariable'. This is not the case here. I request that you keep this and all other instances ('envars', 'stdenv') as they are please.

File: docs/source/markdown/podman-network-reload.1.md
Comment (edsantiago) [reply docs/source/markdown/podman-network-reload.1.md:12]: The preferred way should be 'rootful', one el, because that is the normative English suffix.

File: hack/install_golangci.sh
Comment (edsantiago) [inline hack/install_golangci.sh:15]: As long as you're touching this line, please remove the erroneous apostrophe in `it's`

File: libpod/runtime_ctr.go
Comment (edsantiago) [inline libpod/runtime_ctr.go:346]: Ibid (my "ourself/ourselves" comment above)

File: libpod/container_internal.go
Comment (edsantiago) [inline libpod/container_internal.go:1123]: Although this is not my decision to make, I believe the original usage - implying a singular object - is the correct one. I believe the first-person plural here is the Royal We, which is euphemistically first-person singular. As a reader I find "ourselves" here misleading and confusing; but I will defer to those on the team more familiar with this context.

**Review by edsantiago — CHANGES_REQUESTED** [review]
This is really nice work, and much needed, thank you! I must reject some of your changes as being demonstrably incorrect (envariable, rootfull) and others as possibly so. I would like many more eyeballs on this given the possibility of API- and ABI-breaking changes; and unfortunately the majority of the team is on winter break at the moment. Thank you in advance for your patience, and again an emphatic THANK YOU for taking this on!

File: contrib/cirrus/logformatter
Comment (jsoref) [reply contrib/cirrus/logformatter:443]: Really?
* https://www.google.com/search?q=%22envvariables%22 has About 15,400 results (0.37 seconds) - Did you mean: "env variables"
* https://www.google.com/search?q=%22envariables%22 has About 3,070 results (0.36 seconds)  - Did you mean: "env variables"

I'm not a newbie. I'm happy to drop anything in order to get whatever people will accept, but I don't believe that your assertion about this word is accurate.

**Comment by jsoref** [comment]
Fwiw, you're using both `rootfull` and `rootful`. I guessed wrong based on what felt like dominance. I'll swap them out (it doesn't hurt me, although it's much easier to do before I squash).

If you could give me a heuristic for which things might be ABI breaks, I'd be happy to split them off.

**Comment by jsoref** [comment]
I'm reverting to the unsquashed version for the time being. People can identify commits they'd like me to drop and I'll drop them.

When people are actually ready for this to be merged, can someone authoritatively instruct me to squash? (or you of course have my permission to squash them)

**Comment by zhangguanzhang** [comment]
thansk for contribute, you should rebase all commits to one commit at first

**Comment by jsoref** [comment]
@zhangguanzhang: it's actually hard to respond to requests when people are complaining about individual word sets once the PR is squashed. OTOH, if they complain about a word family and it's still in its component parts, then dropping them is trivial.

Since you aren't going to merge the PR as is at this point, there's really no reason for me to squash it eagerly. It's already cost me time on a few iterations.

As I said above, I'm happy to squash at the end when people are no longer complaining about the actual changes.

**Comment by rhatdan** [comment]
First squash commits and do not change changelogs.

File: vendor/github.com/containers/common/pkg/config/containers.conf
Comment (rhatdan) [inline vendor/github.com/containers/common/pkg/config/containers.conf:421]: Do not change vendored code.

**Review by rhatdan — APPROVED** [review]
I think we should stick to the name "rootfull" rather then "rootful", since neither is a real word, I beleive, and we have lots of blogs, butreports, and other references to rootfull.

**Comment by rhatdan** [comment]
Thanks @jsoref this is a really nice cleanup, and sadly not stuff found by codecleanup.

**Comment by edsantiago** [comment]
>I think we should stick to the name "rootfull" rather then "rootful"

I need to make one last plea: English language rules allow for the use of ["-ful" as a suffix](https://www.etymonline.com/word/-ful), never ["-full"](https://www.etymonline.com/search?q=-full). All instances of "rootfull" are incorrect and jarring; changing existing "rootful" to "rootfull" is making a bad situation much worse. The links above are the result of a quick web search; I'm sorry I don't know how to find definitive references, this is like the how's-the-water fish anecdote: grammar is such a basic deep-down component of my being that I don't really know how/where to find references to fundamental rules. If you need further support on this, please find any language expert you trust and ask them.

>I'm reverting to the unsquashed version for the time being. 

Since this PR may nee further iteration, I'm tentatively OK with @jsoref pushing whatever they prefer during review time, as long as the final version is squashed.

File: contrib/cirrus/logformatter
Comment (edsantiago) [reply contrib/cirrus/logformatter]: Yes, really. Did you look at the results of your searches? In the first few pages of hits for the two-vee form, all (except for a few illiterati, and some MS-DOS) are EnvVariable or envVariable -- mixed case. 

I stand by my assertion. In nearly forty years of UNIX programming, I have never seen the two-vee form used in actual code. That is not per se an arbiter of truth -- I could be wrong, and/or the world may have changed -- but just going by the two links you cite, the one-vee sources are more UNIX-literate (apache, MHonArc, stackexchange) than the all-lower-case two-vee instances ("codegrepper", macports.org).

**Comment by rhatdan** [comment]
Ok if the proper english is ful, then we can go with that. but we need to make sure it is changed everywhere.

**Comment by jsoref** [comment]
@rhatdan: sorry about the `vendor`ed stuff. I usually work w/ a filter that protects from changing `vendor`ed bits:
https://github.com/jsoref/podman/blob/spell-check/.github/actions/spelling/excludes.txt#L10
I probably didn't have it applied (since it isn't part of the PR, it's the extra branch `spell-check` or merge `spelling-with-spell-check`) and I was switching to one of the other repositories. As to not having `codecleanup` catch things. If you like this cleanup, the [action](https://github.com/marketplace/actions/check-spelling) can try to enforce this to protect against future regression.

re: `-ful`/`-full`, I don't really care which way to go (as a native speaker, I do agree that `-ful` should be preferable if it's acceptable, but as a potential contributor, I'll go whichever way the project will let me). I do agree that it should be as consistent as possible, some divergence leads to more divergence.

wrt "everywhere", is there anything else I can do to help? I can apply changes to other public repositories. (This set of PRs -- `buildah`, `podman`, and `skopeo` was added to my queue collectively.) -- Thankfully w/ my tooling, it's just `rs rootfull rootful` which is a really easy thing to do -- although, it's also not much harder for me to create a full spelling PR.

@edsantiago: https://github.com/containers/podman/pull/8787/commits/4d23c731e65b36ba36c2153002f4ed910ee5668d is the current commit for this jitter.

File: libpod/container_config.go
Comment (jsoref) [reply libpod/container_config.go]: @mheon / @rhatdan: is there a style I should use to call this out?

Comment (mheon) [reply libpod/container_config.go]: "The JSON tag is misspelled, but this must be retained for database compatibility" maybe?

**Comment by rhatdan** [comment]
No this is fine, you just need to squash your commits.  When I say everywhere I am talking about mentions outside of the podman repo.

**Comment by rhatdan** [comment]
LGTM
/approve
@containers/podman-maintainers PTAL

**Comment by ashley-cui** [comment]
LGTM, thanks @jsoref!

**Review by QiWang19 — COMMENTED** [review]
LGTM

**Pull request merged**
